ApexOSA/filter.py

```
@@ -0,0 +1,271 @@
+        if isinstance(Mode, str):
+            if Mode.lower() == "dual":
+                Mode = 2
+            else:
+                Mode = 1
+        elif isinstance(Mode, int):
+            if Mode == 2:
+                Mode = 2
+            else:
+                Mode = 1
+        else:
+            raise ApexError(APXXXX_ERROR_ARGUMENT_TYPE, "Mode")
```

Ruff (0.12.2)

`ApexError` may be undefined, or defined from star imports

(F405)